tests/ttnn/unit_tests/gtests/ttnn_test_fixtures.hpp

```diff
@@ -26,4 +29,20 @@ class TTNNFixture : public ::testing::Test {
 
     void TearDown() override { tt::Cluster::instance().set_internal_routing_info_for_ethernet_cores(false); }
 };
+
+class TTNNFixtureWithDevice : public TTNNFixture {
```

Comment:
this helps to make sure that the device is properly closed even if the test throws an unhandled exception.